tools/run_tests/artifacts/build_artifact_python.sh

```diff
@@ -113,6 +113,7 @@ clean_non_source_files() {
 tar xzf "${GRPCIO_TAR_GZ}" -C "${GRPCIO_STRIP_TEMPDIR}"
 ( cd "${GRPCIO_STRIP_TEMPDIR}"
   find . -type d -name .git -exec rm -fr {} \; || true
+  rm -f src/python/grpcio/grpc/_cython/cygrpc.cpp || true
```

Comment:
The content of a source wheel is dictated by [MANIFEST](https://github.com/grpc/grpc/blob/master/PYTHON-MANIFEST.in). I don't see how a cpp file might be included. Can you provide more evidence?

If there is a bug in source wheel file list, I would propose to correct the MANIFEST file.

Reply:
If you look at one of the source distributions in PyPI, you can find the POSIX specific `cygrpc.cpp` which is not useful for windows. E.g: [grpcio-1.44.0.tar.gz](https://files.pythonhosted.org/packages/65/75/8b706e1170e2c7b6242b1675259e47986bb4fc490f29387989a965972e6e/grpcio-1.44.0.tar.gz)

I think extension files are automatically included with the source distribution.

https://packaging.python.org/en/latest/guides/using-manifest-in/

Reply:
Thanks for the official guide pointer, and checking the wheel content. I see the cpp file is got included, and that is on us. The guide didn't mention how extension generated files will get special treatment. Maybe we can try to exclude this file explicitly: https://github.com/grpc/grpc/blob/master/PYTHON-MANIFEST.in#L2

Besides, if we update MANIFEST to fix the problem, we should be able to decouple the building of correct source wheel from this script.

Reply:
Thanks, @lidizheng for the suggestion. I agree that would be a much better fix. I tried it but the files are still not excluded.

change made:

```
+++ b/PYTHON-MANIFEST.in
@@ -1,5 +1,5 @@
 recursive-include src/python/grpcio/grpc *.c *.h *.inc *.py *.pyx *.pxd *.pxi *.python *.pem
-recursive-exclude src/python/grpcio/grpc/_cython *.so *.pyd
+recursive-exclude src/python/grpcio/grpc/_cython *.so *.pyd *.cpp
 graft src/python/grpcio/grpcio.egg-info
 graft src/core
 graft src/boringssl
```

Log:

```
copying src\python\grpcio\grpc\_simple_stubs.py -> grpcio-1.45.0.dev0\src\python\grpcio\grpc
copying src\python\grpcio\grpc\_utilities.py -> grpcio-1.45.0.dev0\src\python\grpcio\grpc
copying src\python\grpcio\grpc/_cython\cygrpc.cpp -> grpcio-1.45.0.dev0\src\python\grpcio\grpc/_cython
```

Either I am doing wrong or `setuptools` ignore it and respect the extension files

Reply:
This indeed feels like a bug (chance to improve `setuptools`). Can we try if `global-exclude` works?

Reply:
I think the exclude issue is specific to windows. It seems to work on my mac. Looks like `setuptools` is struggling with pattern matching for windows (possibly forward slashes etc.). Since we are running the script form a Linux box I think it should be fine to add it to MANIFEST file. I've reverted previous change and added the entry to MANIFEST. I think it should work. but If anyone can try it on their setup and confirm that would be great.